@xen-orchestra/lite/src/components/AddNetwork.tsx

```diff
@@ -0,0 +1,247 @@
+import AddIcon from '@mui/icons-material/Add'
+import React from 'react'
+import SettingsBackupRestoreIcon from '@mui/icons-material/SettingsBackupRestore'
+import { Map } from 'immutable'
+import { SelectChangeEvent } from '@mui/material'
+import { withState } from 'reaclette'
+
+import Button from './Button'
+import Checkbox from './Checkbox'
+import Input from './Input'
+import IntlMessage from './IntlMessage'
+import Select from './Select'
+import { alert } from './Modal'
+
+import XapiConnection, { ObjectsByType, Pif, PifMetrics } from '../libs/xapi'
+
+interface ParentState {
+  objectsByType: ObjectsByType
+  objectsFetched: boolean
+  xapi: XapiConnection
+}
+
+interface State {
+  isBonded: boolean
+  isLoading: boolean
+  form: {
+    [key: string]: unknown
+    bondMode: 'active-backup' | 'balance-slb' | 'lacp' | ''
+    description: string
+    mtu: string
+    nameLabel: string
+    pifsId: string | string[]
+    vlan: string
+  }
+}
+
+interface Props {}
+
+interface ParentEffects {}
+
+interface Effects {
+  createNetwork: React.FormEventHandler<HTMLFormElement>
+  handleChange: (e: SelectChangeEvent<unknown> | React.ChangeEvent<{ name: string; value: unknown }>) => void
+  resetForm: () => void
+  toggleBonded: () => void
+}
+
+interface Computed {
+  collection?: Pif[]
+  pifs?: Map<string, Pif>
+  pifsMetrics?: Map<string, PifMetrics>
+}
+
+const BOND_MODE = ['active-backup', 'balance-slb', 'lacp']
+
+const BUTTON_STYLES = {
+  marginRight: 1,
+  width: 'fit-content',
+}
+
+const OPTION_PIF_RENDERER = (pif: Pif, { pifsMetrics }: { pifsMetrics: Computed['pifsMetrics'] }) =>
+  `${pif.device} (${pifsMetrics?.find(metrics => metrics.$ref === pif.metrics)?.device_name ?? 'unknown'})`
+
+const INPUT_STYLES = {
+  marginBottom: 2,
+}
+
+const AddNetwork = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      isBonded: false,
+      isLoading: false,
+      form: {
+        bondMode: '',
+        description: '',
+        mtu: '',
+        nameLabel: '',
+        pifsId: '',
+        vlan: '',
+      },
+    }),
+    computed: {
+      pifs: state => state.objectsByType.get('PIF'),
+      pifsMetrics: state => state.objectsByType.get('PIF_metrics'),
+      collection: state =>
+        state.pifs
+          ?.filter(pif => pif.VLAN === -1 && pif.bond_slave_of === 'OpaqueRef:NULL' && pif.host === pif.$pool.master)
+          .sortBy(pif => pif.device)
+          .valueSeq()
+          .toArray(),
+    },
+    effects: {
+      createNetwork: async function (e) {
+        e.preventDefault()
+        if (this.state.isLoading) {
+          return
+        }
+        this.state.isLoading = true
```

Comment:
Why do you need this?

Reply:
To avoid sending multiple request to the server simultaneous and to disabled the button for a better UX, if a request is already handle by the server.

Reply:
We can create an `ActionButton` to handle this behaviour. But it's out of the scope of this PR

Reply:
You can make a separate PR, as it will be useful later on.

ping @pdonias

Reply:
Yes, we can create an action button eventually. In the meantime, this is ok. You can leave a comment saying that this will be handled by the button and that we should remove this once it's done.